internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go

```diff
@@ -730,40 +724,30 @@ func (s *service) Stat(ctx context.Context, req *provider.StatRequest) (*provide
 		}, nil
 	}
 
-	path := req.Ref.Path
-	if receivedShare.MountPoint.Path == strings.TrimPrefix(req.Ref.Path, "./") {
-		path = "."
-	}
-
 	// TODO return reference?
 	return s.gateway.Stat(ctx, &provider.StatRequest{
-		Opaque: req.Opaque,
-		Ref: &provider.Reference{
-			ResourceId: receivedShare.Share.ResourceId,
-			Path:       path,
-		},
+		Opaque:                req.Opaque,
+		Ref:                   buildReferenceInShare(req.Ref, receivedShare),
 		ArbitraryMetadataKeys: req.ArbitraryMetadataKeys,
 	})
+
+	// we need to rewrite the id if the request was made relative to the virtual space root?
+	// but won't that be problematic for eg. wopi because it needs the correct id?
+	// the web ui seems to continue navigating based on the id.
```

Comment:
Please add more clear to call to action. Is it a TODO? If yes, what needs to be done? Or is a NOTE? FIXME? IAMATEAPOT?